tools/validators/instance_validator/validate/entity_instance.py

```diff
@@ -126,9 +128,15 @@ def _ValidateLinks(self):
 
       # scan all standard fields and ensure they're defined
       fields_map = dict(links[entity_name])
-      field_universe = self.universe.field_universe
+      source_entity_instance = entity_instances.get(entity_name)
+      # this is valid type, the check already happened
+      entity_type_str = str(source_entity_instance.entity['type'])
+      type_parse = entity_type_str.split('/')
+      namespace = type_parse[0]
+      entity_type = type_parse[1]
+      entity_type_universe = self.universe.GetEntityType(namespace, entity_type)
       for sourcename in fields_map.keys():
-        if not field_universe.IsFieldDefined(fields_map[sourcename], ''):
+        if not entity_type_universe.HasField('/'+sourcename.data):
```

Comment:
add a comment indicating the assumption here (since it's not necessarily true forever)